### DISCUSSION TITLE
Return typings for RPCs with the JavaScript-client

### DISCUSSION THREAD

**Discussion by ameesme** [supabase/supabase#10464] [Questions] [answered]
We are using Supabase with a couple of RPCs and noticed that TypeScript return-types are defined as `Record<string,unknown>[]` instead of the actual return-values described in the Postgres function-definition.

Is this expected behavior?

### Postgres Function
```sql
CREATE OR REPLACE FUNCTION public.job_get_recommendations(
	skills bigint[],
	weekly_hours integer,
	location_geo json,
	travel_distance_kilometers integer
) RETURNS TABLE(
 	id integer,
 	title character varying,
 	description character varying,
 	skills bigint[],
 	travel_distance_kilometers integer,
 	location_geo json,
 	score integer,
 	reason recommendation_reasons[]
)
```

### Generated Typescript-definition
```js
job_get_recommendations: {
  Args: {
    skills: unknown
    weekly_hours: number
    location_geo: Json
    travel_distance_kilometers: number
  }
  Returns: Record<string, unknown>[]
}
```

**Comment by Rutger** [comment]
I'm experiencing this same issue. I must be missing something.

**Comment by steve-chavez** [comment]
Linking https://github.com/supabase/postgrest-js/issues/374

**Comment by soedirgo** [comment, marked as answer]
This has been resolved by https://github.com/supabase/postgres-meta/pull/423
